Factory reset

mook-as opened the pull request:
This adds a factory reset functionality, that will delete all configuration and restart the app.

Fixes #37.

mattfarina commented:
@mook-as can you rebase? I merged the dashboard PR and a rebase is needed now.

mattfarina reviewed (commented):
I've got some feedback...

After hitting the reset button the application restarted but when it came back up the preferences showed a white screen. This is through running it with `npm run dev`. It did restart. It just didn't come back up right. I tried to run it as a built app but that just caused it to come up in error.

When I did bring it back up (after the built app tests) using `npm run dev` it came back up with running k8s but the factor reset button can't be clicked. See below...

<img width="527" alt="Screen Shot 2021-01-15 at 11 49 32 AM" src="https://user-images.githubusercontent.com/62991/104754958-d2fbf200-5727-11eb-9e71-a2417d8d8a17.png">

In fact, for some reason I can get k8s into a started state but the factor reset button never comes in as usable again.

mook-as commented:
- Rebased on top of `main`.
- Fixed the issue with the reset button being disabled: we now have an additional `READY` state (from the dashboard work) from where we should be allowed to reset.
- I don't think `npm run dev` will work: when we restart the app, the controlling process (the thing that actually runs `npm run electron:serve`) sees that electron has exited, and kills the dev server running the vue compilation.  So when the new electron starts up, it can't talk to the server (on localhost), causing the load to fail.  Electron happens to have a completely blank screen for that instead of a proper chrome (dinosaur) error page.

Running `open dist/electron/mac/rancher-desktop.app` does make things work for me on factory reset.
